src/adapter/threads.ts

```diff
@@ -902,26 +902,38 @@ export class Thread implements IVariableStoreDelegate {
       event.args[0].value = localize('console.assert', 'Assertion failed');
 
     let messageText: string;
+    let usedAllArgs = false;
     if (event.type === 'table' && event.args.length && event.args[0].preview) {
       messageText = objectPreview.formatAsTable(event.args[0].preview);
     } else {
       const useMessageFormat = event.args.length > 1 && event.args[0].type === 'string';
       const formatString = useMessageFormat ? (event.args[0].value as string) : '';
-      messageText = messageFormat.formatMessage(
+      const formatResult = messageFormat.formatMessage(
         formatString,
         useMessageFormat ? event.args.slice(1) : event.args,
         objectPreview.messageFormatters,
       );
+      messageText = formatResult.result;
+      usedAllArgs = formatResult.usedAllSubs;
+    }
+
+    let output: string;
+    let variablesReference: number | undefined;
+    if (!usedAllArgs || event.args.some(arg => objectPreview.isObject(arg))) {
```

Comment:
`isObject` excludes several [subtypes](https://github.com/microsoft/vscode-js-debug/blob/master/src/adapter/objectPreview.ts#L17). You probably want a different check here.

Reply:
So this is what VariableStore uses to decide whether to send a variable for an arg, or just a text preview, and that's why I used it, but I think that logic is wrong because currently we can't inspect the properties of those types at all, and you can in chrome devtools. So I will leave this and file an issue for handing those types correctly.

https://github.com/microsoft/vscode-js-debug/blob/9af83b87b156f97cd70d99cd4742fdb04f56f893/src/adapter/variables.ts#L252-L260

![image](https://user-images.githubusercontent.com/323878/72365250-b3616200-36ac-11ea-9555-35b7fee2a4f7.png)

Reply:
Ok